prov/net/src/xnet_progress.c

```diff
@@ -237,12 +345,17 @@ static void xnet_progress_tx(struct xnet_ep *ep)
 	ssize_t ret;
 
 	assert(xnet_progress_locked(xnet_ep2_progress(ep)));
-	while (ep->cur_tx.entry) {
+	while (ep->cur_tx.entry && !ep->cur_tx.io_uring_busy) {
```

Comment:
I need to look at how io_uring_busy is used.  I mostly follow what you're doing here.  But I want to understand where the flow differs between using io_uring and async sends and why we have those differences.